### PR TITLE
[Feature] Add Experience Gain Toggle.

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Perl
- Add `$client->IsEXPEnabled()`.
- Add `$client->SetEXPEnabled(is_exp_enabled)`.

# Lua
- Add `client:IsEXPEnabled()`.
- Add `client:SetEXPEnabled(is_exp_enabled)`.

# Commands
- Add `#exptoggle [Toggle] - Toggle your or your target's experience gain.`.

# Notes
- Allows operators to turn on/off a player's experience gain individually without changing their rule values.
- The command allows operators to give players access to the command to disable their own experience gain.

### CODE REVIEW

File: zone/command.cpp
Comment (Akkadius) [inline zone/command.cpp:362]: Use full word “exp” vs “xp”

Comment (Kinglykrab) [reply zone/command.cpp:362]: Sounds good.

**Comment by Akkadius** [comment]
I’ll look closer tomorrow, about to 😴

**Review by Aeadoin — APPROVED** [review]
Not seeing any issues, only thing I was thinking was that exp_enabled would need to be initialized in the client constructor... but that's taken care of via: 

`memset(&m_epp, 0, sizeof(m_epp));` 

Will let Akk take another look before merging.

File: common/extprofile.h
Comment (Akkadius) [inline common/extprofile.h:59]: There's really no good reason why this should live here. This can simply be a private member on `Client` and the setters/getters can mutate it 

eg

```cpp
private:
bool m_exp_enabled = {};
```

File: zone/client.cpp
Comment (Akkadius) [inline zone/client.cpp:9155]: Update to feedback above

File: zone/gm_commands/exptoggle.cpp
Comment (Akkadius) [inline zone/gm_commands/exptoggle.cpp:27]: Clean

File: zone/zonedb.cpp
Comment (Akkadius) [inline zone/zonedb.cpp:1383]: We won't have access to the client object in this method so lets just pass the client pointer for now and pull things out of the client object within this save routine.

I'd avoid converting this to repositories for now

All of this client save stuff needs to get cleaned up

Comment (Kinglykrab) [reply zone/zonedb.cpp:1383]: Yeah the reason I put it in extended profile is that we didn’t pass anything but player profile and extended profile to this method.

File: common/extprofile.h
Comment (Kinglykrab) [reply common/extprofile.h:59]: Reason was we don’t pass client object to save method.

**Review by Akkadius — APPROVED** [review]
LGTM; assuming the character save changes were thoroughly tested

**Pull request merged**
